### ISSUE TITLE
Regarding the implementation of the Alignment loss

### ISSUE THREAD

**Issue by changliu816** [sunyiyou/SAFAS#13]
Hi Yiyou,

Congrats on your paper acceptance! Great work. 

I think I have trouble finding your implementation on alignment loss in Eqn. 5 in the paper.  I also checked here:           
  loss_all = cls_loss + args.feat_loss_weight * feat_loss at line 191, train.py
where I did not find the alignment loss. Please let me know if I missed anything.

Best,
Chang

**Comment by sunyiyou** [comment]
Hi Chang, 

Thanks for your interest in our work! 

Our alignment is given by the IRM objective. In our paper, we converted it into a projected gradient descent algorithm. 

Specifically, it is in Line 9 of algorithm 1, which is quite easy to implement.
